test/residency/prototypes/test_replay_modes_public_seam.py

```
@@ -0,0 +1,688 @@
+
+    def test_task014_native_compile_runs_in_temporary_directory(self):
+        relative_path, _ = SEAMS["TASK-014"]
+        seam = load_module(
+            ROOT / relative_path,
+            "task014_native_compile_directory_test",
+        )
+        source = ROOT / "test/cpp/test_residency_prototype_task014.cpp"
+
+        class ReplayContract:
+            @staticmethod
+            def resolve_replay_compiler(_observation, _platform_id, _environment):
```

Ruff (0.16.1)

[warning] Missing return type annotation for staticmethod `resolve_replay_compiler`

(ANN205)